service/tasks/machine.py

```diff
@@ -197,27 +195,41 @@ def enable_image_validation(machine_request, init_task, final_task, original_sta
     else:
         validate_task = validate_new_image.s(machine_request.id)
         init_task.link(validate_task)
-    #Validate task returns an instance_id
-    # Task 4 = Wait for new instance to be 'active'
-    wait_for_task = wait_for_instance.s(
-        # NOTE: 1st arg, instance_id, passed from last task.
+    validate_task.link(destroy_task)
+    validate_task.link_error(error_handler_task)
+    # Task 4 = Terminate the new instance on completion
+    destroy_task = destroy_instance_once_active.s(
         admin_driver.__class__,
         admin_driver.provider,
         admin_driver.identity,
-        "active",
-        test_tmp_status=True,
-        return_id=True)
-    validate_task.link(wait_for_task)
-    validate_task.link_error(error_handler_task)
-    # Task 5 = Terminate the new instance on completion
-    destroy_task = destroy_instance.s(
```

Comment:
wait_for_instance was forced to return an instance uuid just so destroy instance could get a param passed to it.